commands/cloudapi-v6/natgateway.go

```diff
@@ -274,21 +282,30 @@ func RunNatGatewayUpdate(c *core.CommandConfig) error {
 }
 
 func RunNatGatewayDelete(c *core.CommandConfig) error {
-	if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete nat gateway"); err != nil {
-		return err
-	}
+	var resp *resources.Response
 	dcId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgDataCenterId))
 	natGatewayId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgNatGatewayId))
-	c.Printer.Verbose("NatGateway with id: %v is deleting...", natGatewayId)
-	resp, err := c.CloudApiV6Services.NatGateways().Delete(dcId, natGatewayId)
-	if resp != nil {
-		c.Printer.Verbose(cloudapiv6.RequestTimeMessage, resp.RequestTime)
-	}
-	if err != nil {
-		return err
-	}
-	if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
-		return err
+	allFlag := viper.GetBool(core.GetFlagName(c.NS, cloudapiv6.ArgAll))
+	if allFlag {
+		err := DeleteAllNatgateways(c)
+		if err != nil {
+			return err
+		}
+	} else {
+		if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete nat gateway"); err != nil {
+			return err
+		}
+		c.Printer.Verbose("Starring deleting NatGateway with id: %v...", natGatewayId)
+		resp, err := c.CloudApiV6Services.NatGateways().Delete(dcId, natGatewayId)
```

Comment:
Use = instead of := to not define resp as a new variable. It will not be recognized at the end of the current function.